core/src/main/kotlin/com/strumenta/starlasu/transformation/Transformation.kt

```diff
@@ -158,30 +167,42 @@ class NodeFactory<Source, Output : ASTNode>(
         name: String,
         scopedToType: KClass<*>? = null,
         childType: KClass<out ASTNode> = ASTNode::class,
-    ): NodeFactory<Source, Output> {
+    ): Transform<Source, Output> {
+        if (childrenPolicy == ChildrenPolicy.SKIP) {
```

Comment:
Should we handle the different children policy through a top-level when statement?